core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -205,38 +222,39 @@ class UserService(protected val ctx: RawlsRequestContext,
       }
     } yield constructBillingProjectResponseFromOptionalAndRoles(maybeBillingProject, projectRoles)
 
-  private def makeBillingProjectResponse(projectRoles: Set[ProjectRole], billingProject: RawlsBillingProject) =
-    RawlsBillingProjectResponse(
-      billingProject.projectName,
-      billingProject.billingAccount,
-      billingProject.servicePerimeter,
-      billingProject.invalidBillingAccount,
-      projectRoles,
-      billingProject.status,
-      billingProject.message,
-      billingProject.azureManagedAppCoordinates
-    )
-
-  def listBillingProjectsV2(): Future[List[RawlsBillingProjectResponse]] =
-    for {
-      samUserResources <- samDAO.listUserResources(SamResourceTypeNames.billingProject, ctx.userInfo)
-      projectNames = samUserResources.map(r => RawlsBillingProjectName(r.resourceId)).toSet
-      projectsInDB <- dataSource.inTransaction { dataAccess =>
-        dataAccess.rawlsBillingProjectQuery.getBillingProjects(projectNames)
-      }
-      bpmProfiles <- billingProfileManagerDAO.listBillingProfiles(samUserResources, ctx)
-    } yield constructBillingProjectResponses(samUserResources, projectsInDB ++ bpmProfiles)
-
-  private def constructBillingProjectResponses(samUserResources: Seq[SamUserResource],
-                                               billingProjectsInRawlsDB: Seq[RawlsBillingProject]
-  ): List[RawlsBillingProjectResponse] = {
-    val projectsByName = billingProjectsInRawlsDB.map(p => p.projectName.value -> p).toMap
-    samUserResources.toList
-      .flatMap { samUserResource =>
-        val projectRoles = samRolesToProjectRoles(samUserResource.direct.roles ++ samUserResource.inherited.roles)
-        projectsByName.get(samUserResource.resourceId).map(makeBillingProjectResponse(projectRoles, _))
-      }
-      .sortBy(_.projectName.value)
+  def listBillingProjectsV2(): Future[List[RawlsBillingProjectResponse]] = for {
```

Comment:
I completely refactored this, when I added error handling for when a billing profile isn't found.

I started by nesting constructBillingProjectResponses inside listBillingProjectsV2, since it was only used there, but then I realized the resources it used were available for the entire sequence of transformations, if I just put the call to sam at the top.

This let me transform more directly to the RawlsBillingProjectResponse, and indicate the lack of a billing profile there.